### PR TITLE
Bagel Voting Requirement

### PR DESCRIPTION
**Pull request by Kurfursten**
Adds bagels, cook them the same as donuts but butter instead of eggs (5 flour 1 butter)

In order to vote for Bagelstation, 17 of them have to be cooked in one shift, a nod to the fact that Bagel was originally revived in 2017

🆑 
* rscadd: Bagels can now be cooked.
* tweak: Bagelstation requires 17 bagels to be cooked to appear on the map vote.

### CODE REVIEW

File: code/__HELPERS/files.dm [hunk 1]
```diff
@@ -1,3 +1,5 @@
+#define BAGEL_REQUIREMENT 17
```
Comment (DamianX) [inline code/__HELPERS/files.dm:1]: `#undef` at the end of the file

**Comment by DrSnips** [comment]
I get it's quirky and funny but I'm not so sure about seeing silly "map requirements" becoming more common. This is going to get out of hand soon mark my words.

**Comment by Exxion** [comment]
They're also kind of unpleasant codewise, tying code to things that should be config. But there's no easy way to make requirements like these into config, beyond superficially.

**Comment by Kurfursten** [comment]
@DrSnips I actually like them, because I think they avoid what I'm now calling "map sprawl" where we have so many choices that we have 5 votes box, 3 votes meta, 2 votes deff, 2 votes synergy, 2 votes bagel, 1 vote snaxi, 3 votes castle, 2 votes lamprey.

By adding a requirement, it's more special and more likely to consolidate votes when it is available. And at the same time, when it doesn't meet the requirement, it will not be draining votes away from the anti-box alliance.

@Exxion More like Byond superficially

**Comment by SonixApache** [comment]
make it 2017 bagels in one round and you've got a deal

**Pull request merged**